include/libopencm3/stm32/h7/flash.h

```diff
@@ -27,27 +27,53 @@
 #define LIBOPENCM3_FLASH_H
 
 #include <libopencm3/stm32/common/flash_common_all.h>
-#include <libopencm3/stm32/common/flash_common_f.h>
-#include <libopencm3/stm32/common/flash_common_f24.h>
 
 /**@{*/
+/** @defgroup flash_registers Flash Registers
+ * @ingroup flash_defines
+@{*/
+/** Flash Access Control register */
+#define FLASH_ACR                     MMIO32(FLASH_MEM_INTERFACE_BASE + 0x00)
+/** Flash Key register */
+#define FLASH_KEYR                    MMIO32(FLASH_MEM_INTERFACE_BASE + 0x04)
+/** Flash Option bytes key register */
+#define FLASH_OPTKEYR                 MMIO32(FLASH_MEM_INTERFACE_BASE + 0x08)
+/*@}*/
 
-/** @addtogroup flash_acr_values FLASH_ACR values
- * @ingroup flash_registers
+/** @defgroup flash_latency FLASH Wait States
+@ingroup flash_defines
 @{*/
-#define FLASH_ACR_WRHF_VOS1_70MHZ     (0 << FLASH_ACR_WRHIGHFREQ_SHIFT)
-#define FLASH_ACR_WRHF_VOS1_185MHZ    (1 << FLASH_ACR_WRHIGHFREQ_SHIFT)
-#define FLASH_ACR_WRHF_VOS1_225MHZ    (2 << FLASH_ACR_WRHIGHFREQ_SHIFT)
-#define FLASH_ACR_WRHF_VOS2_55MHZ     (0 << FLASH_ACR_WRHIGHFREQ_SHIFT)
-#define FLASH_ACR_WRHF_VOS2_165MHZ    (1 << FLASH_ACR_WRHIGHFREQ_SHIFT)
-#define FLASH_ACR_WRHF_VOS2_225MHZ    (2 << FLASH_ACR_WRHIGHFREQ_SHIFT)
-#define FLASH_ACR_WRHF_VOS3_45MHZ     (0 << FLASH_ACR_WRHIGHFREQ_SHIFT)
-#define FLASH_ACR_WRHF_VOS3_135MHZ    (1 << FLASH_ACR_WRHIGHFREQ_SHIFT)
-#define FLASH_ACR_WRHF_VOS3_225MHZ    (2 << FLASH_ACR_WRHIGHFREQ_SHIFT)
+#define FLASH_ACR_LATENCY(w)          ((w) & FLASH_ACR_LATENCY_MASK)
+#define FLASH_ACR_LATENCY_0WS         0x00
+#define FLASH_ACR_LATENCY_1WS         0x01
+#define FLASH_ACR_LATENCY_2WS         0x02
+#define FLASH_ACR_LATENCY_3WS         0x03
+#define FLASH_ACR_LATENCY_4WS         0x04
+#define FLASH_ACR_LATENCY_5WS         0x05
+#define FLASH_ACR_LATENCY_6WS         0x06
+#define FLASH_ACR_LATENCY_7WS         0x07
+#define FLASH_ACR_LATENCY_SHIFT       0
+#define FLASH_ACR_LATENCY_MASK        0x0f
+#define FLASH_ACR_WRHIGHFREQ_MASK     0x3
+#define FLASH_ACR_WRHIGHFREQ_SHIFT    4
 /*@}*/
-#define FLASH_ACR_WRHIGHFREQ_MASK     (0x3)
-#define FLASH_ACR_WRHIGHFREQ_SHIFT    (0x4)
 
+/** @defgroup flash_acr_values FLASH_ACR values
+ * @ingroup flash_registers
+ * @brief Access Control register values
+ * @{*/
+/* Unsupported, make this operation a no-op. */
+#define FLASH_ACR_PRFTEN              (0)
```

Comment:
which common api is trying to use this that breaks without this define?  It seems cute, but I'm a little surprised it maybe surprising.

Reply:
flash_prefetch_enable()/disable()